### PR TITLE
extened embedlyInfo and s3 store

### PR DESCRIPTION
**Pull request by yingjieMiao**
@raymondng  @martinraison 

It has 4 additional fields: uriId, entities, keywords, calledEmbedlyAt

We can construct EmbedlyInfo from this extended Info. Only ExtendedEmbedlyInfo is persisted in S3. 
Nothing will be changed in UriSummaryCommander.

### CODE REVIEW

**Comment by martinraison** [comment]
looks good. Are keywords and entities big? I was trying to figure out if we really need two distinct types of EmbedlyInfo objects (normal one and extended one).

**Comment by yingjieMiao** [comment]
the lists are sometimes long. (for long articles)

I agree, we can gradually retire the old format.

I will update this pull request

**Comment by raymondng** [comment]
Just checking: are the two elements guaranteed to exist?

**Comment by yingjieMiao** [comment]
updated, added a test

Ray: the two elements are Seq, so Seq() if embedly returns nothing. It's identical to how we handle images.

File: kifi-backend/modules/scraper/test/data/sample_embedly.json [hunk 1]
```diff
@@ -0,0 +1 @@
...out sacrifices (battery life was excellent, for instance, especially when using the smartphone's display sparingly). The camera is a decent performer, beating the Nexus 5 by a wide margin, though falling short of the Galaxy S5 and the iPhone in this department. But its flaws are small things, details that fail by matters of degrees rather than huge margins, and for a smartphone that costs less than half of many of its competitors, that's nothing short of amazing.</p>\n</div>", "entities": [{"count": 4, "name": "Google"}, {"count": 1, "name": "RAM"}, {"count": 1, "name": "Software Speaking of Cyanogen"}, {"count": 1, "name": "Samsung Galaxy"}], "favicon_colors": [{"color": [0, 148, 1], "weight": 0.208740234375}, {"color": [252, 252, 252], "weight": 0.041259765625}], "keywords": [{"score": 120, "name": "oneplus"}, {"score": 66, "name": "devices"}, {"score": 53, "name": "nexus"}, {"score": 44, "name": "android"}, {"score": 33, "name": "display"}, {"score": 31, "name": "smartphone"}, {"score": 30, "name": "5-inch"}, {"score": 29, "name": "screen"}, {"score": 28, "name": "specs"}, {"score": 24, "name": "phone"}], "published": 1400586550000, "provider_name": "TechCrunch", "type": "html"}
```
Comment (martinraison) [inline kifi-backend/modules/scraper/test/data/sample_embedly.json]: I suggest formatting it nicely (for example with http://jsonviewer.stack.hu/) to make it easier to debug the tests.

**Comment by martinraison** [comment]
looks good (just made a small comment)

File: kifi-backend/modules/scraper/test/data/sample_embedly.json (hunk 1 above)
Comment (yingjieMiao) [inline kifi-backend/modules/scraper/test/data/sample_embedly.json]: cool

**Comment by raymondng** [comment]
Cool. I was wondering about whether they are the same as images (i.e.
always there, even if empty) -- if that's the case, then it's good.

File: kifi-backend/modules/common/app/com/keepit/scraper/embedly/EmbedlyStore.scala [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import com.keepit.common.logging.AccessLog
+import com.keepit.common.store.{InMemoryObjectStore, ObjectStore, S3Bucket, S3JsonStore}
+import com.keepit.model.NormalizedURI
+
+import play.api.libs.json.Format
+
+
+trait EmbedlyStore extends ObjectStore[Id[NormalizedURI], StoredExtendedEmbedlyInfo]
+
+class S3EmbedlyStoreImpl(val bucketName: S3Bucket, val amazonS3Client: AmazonS3, val accessLog: AccessLog, val formatter: Format[StoredExtendedEmbedlyInfo] = StoredExtendedEmbedlyInfo.format)
```
Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/scraper/embedly/EmbedlyStore.scala:15]: Nice :+1:

**Comment by raymondng** [comment]
BTW don't see any S3 config changes -- are these already taken care of?

**Comment by yingjieMiao** [comment]
@ray,

s3 config will be next. I tried to modularize the commits.

**Comment by yingjieMiao** [comment]
@martinraison 

json formatted. Thanks for the tool. merging.

**Comment by martinraison** [comment]
cool, thanks!

**Pull request merged**